tasks/extract-binaries-from-image/extract-binaries-from-image.yaml

```diff
@@ -41,28 +45,48 @@ spec:
             echo "Error: No valid snapshot file was provided."
             exit 1
         fi
-        IMAGE_URL=$(jq -r '.components[0].containerImage // ""' "${SNAPSHOT_SPEC_FILE}")
-        if [ -z "${IMAGE_URL}" ] ; then
-            echo "Error: Unable to get image url from snapshot."
-            exit 1
-        fi
 
         BINARIES_DIR=binaries
         BINARIES_PATH=$(workspaces.data.path)/$(params.subdirectory)/$BINARIES_DIR
         mkdir -p $BINARIES_PATH
 
-        TMP_DIR=$(mktemp -d)
-        skopeo copy docker://$IMAGE_URL dir:$TMP_DIR
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            DESIRED_COMPONENTS_LIST=()
+        else
+            DESIRED_COMPONENTS_LIST=$(jq -r '.["content-gateway"].components[].name' "${DATA_FILE}")
```

Comment:
I think the `[]` in `["content-gateway"]` are redundant. But also, if `content-gateway` is not even defined in the data file, this will fail:
```sh
$ jq '."content-gateway".components[]' <<< '{}'
jq: error (at <stdin>:1): Cannot iterate over null (null)
```

Even if it's there, but has no components:
```sh
$jq '."content-gateway".components[]' <<< '{"content-gateway": {}}'
jq: error (at <stdin>:1): Cannot iterate over null (null)
```

So you should probably first check if the key is there:
```sh
elif [ $(jq '."content-gateway" | has("components")' "${DATA_FILE}") = true ]; then
```

Reply:
That condition should be covered by this schema https://issues.redhat.com/browse/CLOUDDST-22190.

It's probably good to have the check here too as you pointed out. Added, thanks!